stdlib/builtins.pyi

```diff
@@ -940,24 +939,35 @@ class bool(int):
     def __invert__(self) -> int: ...
 
 @final
-class slice(Generic[_StartT, _StopT, _StepT]):
+class slice(Generic[_StartT_co, _StopT_co, _StepT_co]):
     @property
-    def start(self) -> _StartT: ...
+    def start(self) -> _StartT_co: ...
     @property
-    def step(self) -> _StepT: ...
+    def step(self) -> _StepT_co: ...
     @property
-    def stop(self) -> _StopT: ...
-    @overload
-    def __new__(cls, stop: int | None, /) -> slice[int | MaybeNone, int | MaybeNone, int | MaybeNone]: ...
-    @overload
-    def __new__(
-        cls, start: int | None, stop: int | None, step: int | None = None, /
-    ) -> slice[int | MaybeNone, int | MaybeNone, int | MaybeNone]: ...
-    @overload
-    def __new__(cls, stop: _T2, /) -> slice[Any, _T2, Any]: ...
-    @overload
-    def __new__(cls, start: _T1, stop: _T2, /) -> slice[_T1, _T2, Any]: ...
-    @overload
+    def stop(self) -> _StopT_co: ...
+    # unary overloads ------------------------------------------------------------------
+    @overload  # 1x None
+    def __new__(cls, stop: None, /) -> slice: ...
+    @overload  # 0x None
+    def __new__(cls, stop: _T2, /) -> slice[_T2 | None]: ...
```

Comment:
I know it's not _necessary_, because you have `_StopT` defaulting to whatever `_StartT` is, but I'd find it clearer to specify both here, given that `stop` is the only value given:

```suggestion
    def __new__(cls, stop: _T2, /) -> slice[_T2 | None, _T2 | None]: ...
```